libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -387,16 +392,22 @@ Compass::Compass(void) :
     }
 }
 
+//pass reference to compass singleton
+Compass& Compass::get_frontend(void)
```

Comment:
I would follow what we already have in AP_InertialSensor() and which is common idiom in other singleton implementations:  get_instance()